--- linux-2.6.31/arch/mips/atheros/gpio.c	2014-09-01 08:38:01.422383245 -0700
+++ linux-2.6.31-new/arch/mips/atheros/gpio.c	2014-09-01 08:37:48.610353501 -0700
@@ -1273,7 +1273,9 @@ int __init ath_simple_config_init(void)
     if (req3!= 0) {
         printk("request_irq for sw reset/factory reset failed (error %d)\n", req3);
         misc_deregister(&athfr_miscdev);
+#ifdef JUMPSTART_GPIO
         free_irq(req, NULL);
+#endif
         return -1;
     }
     printk("req_num: %d\n", req3);
--- linux-2.6.31/arch/mips/atheros/Makefile	2014-09-01 08:26:25.838352994 -0700
+++ linux-2.6.31-new/arch/mips/atheros/Makefile	2014-09-01 08:25:59.342354188 -0700
@@ -71,7 +71,7 @@ ifneq ($(CONFIG_JUMPSTART_GPIO_SUPPORT),
 JUMPSTART_GPIO=$(CONFIG_JUMPSTART_GPIO_NUMBER)
 else
 ifndef CONFIG_MACH_QCA953x 
-JUMPSTART_GPIO=12
+#JUMPSTART_GPIO=12  #disable jumpstart
 endif
 endif
 endif
